BookingProject/BookingWebAPI/BookingWebApi.Infrastructure/Data/AppUserRepository.cs

```diff
@@ -131,5 +134,15 @@ public async Task<Result<UserInfo>> GetUserInfoById(string userId)
 
             return Result<UserInfo>.Success(result);    
         }
+
+        public async Task<string> GetRoleById(string userId)
+        {
+            var user = await _context.Users
+                .FirstOrDefaultAsync(x => x.Id == userId);
+
+            var roles = await _userManager.GetUserRoles(user);
```

Comment:
U have access to context, so u don't need to make 2 queries. Just make join, like:
select r.Name from users join userRoles join roles r where userId =your_id

Reply:
and u can remove decorator from dependencies